File: app/src/main/java/com/dpis/module/templates/TemplateEditorDraft.java
package com.dpis.module.templates;

import com.dpis.module.appconfig.AppConfigInputValidation;
import com.dpis.module.fonts.FontApplyMode;
import com.dpis.module.viewport.ViewportApplyMode;
import com.dpis.module.viewport.ViewportTargetType;

/** Cross-configuration draft seed shared by the Compose template editor surfaces. */
public final class TemplateEditorDraft {
    public final boolean quickTemplate;
    public final String nameInput;
    public final String viewportInput;
    public final String viewportMode;
    public final String viewportApplyMode;
    public final String viewportScaleInput;
    public final String viewportAbsoluteInput;
    public final String fontInput;
    public final String fontMode;
    public final String selectedTypefaceId;
    public final String draftFontHookDomainsRaw;

    public TemplateEditorDraft(
            boolean quickTemplate,
            String nameInput,
            String viewportInput,
            String viewportMode,
            String viewportApplyMode,
            String viewportScaleInput,
            String viewportAbsoluteInput,
            String fontInput,
            String fontMode,
            String selectedTypefaceId,
            String draftFontHookDomainsRaw
    ) {
        this.quickTemplate = quickTemplate;
        this.nameInput = text(nameInput);
        this.viewportInput = text(viewportInput);
        this.viewportMode = ViewportTargetType.normalize(viewportMode);
        this.viewportApplyMode = ViewportApplyMode.normalize(viewportApplyMode);
        this.viewportScaleInput = text(viewportScaleInput);
        this.viewportAbsoluteInput = text(viewportAbsoluteInput);
        this.fontInput = text(fontInput);
        this.fontMode = AppConfigInputValidation.initialFontMode(fontMode);
        this.selectedTypefaceId = selectedTypefaceId;
        this.draftFontHookDomainsRaw = draftFontHookDomainsRaw;
    }

    private static String text(String value) {
        return value != null ? value : "";
    }
}

File: app/src/main/java/com/dpis/module/templates/TemplateEditorDraft.kt
package com.dpis.module.templates

import com.dpis.module.appconfig.AppConfigInputValidation
import com.dpis.module.fonts.FontApplyMode
import com.dpis.module.viewport.ViewportApplyMode
import com.dpis.module.viewport.ViewportTargetType

/** Cross-configuration draft seed shared by the Compose template editor surfaces. */
class TemplateEditorDraft(
    @JvmField val quickTemplate: Boolean,
    nameInput: String?,
    viewportInput: String?,
    viewportMode: String?,
    viewportApplyMode: String?,
    viewportScaleInput: String?,
    viewportAbsoluteInput: String?,
    fontInput: String?,
    fontMode: String?,
    @JvmField val selectedTypefaceId: String?,
    @JvmField val draftFontHookDomainsRaw: String?,
) {
    @JvmField val nameInput = nameInput.orEmpty()
    @JvmField val viewportInput = viewportInput.orEmpty()
    @JvmField val viewportMode = ViewportTargetType.normalize(viewportMode)
    @JvmField val viewportApplyMode = ViewportApplyMode.normalize(viewportApplyMode)
    @JvmField val viewportScaleInput = viewportScaleInput.orEmpty()
    @JvmField val viewportAbsoluteInput = viewportAbsoluteInput.orEmpty()
    @JvmField val fontInput = fontInput.orEmpty()
    @JvmField val fontMode = AppConfigInputValidation.initialFontMode(fontMode)
}
